fta/samples/api/views.py

```diff
@@ -56,16 +56,40 @@ def add_fathom_sample(self, request, format=None):
         freeze_software = request.data["freeze_software"]
         notes = request.data["notes"] if "notes" in request.data else ""
         sample = sample_from_required(frozen_page, freeze_software, notes)
-        sample.save()
+        existing_sample = Sample.objects.filter(url__contains=sample.url).order_by(
```

Comment:
Actually thinking more about this I'm not sure we can get by with just filtering on content like this, even with an exact match, because multiple `Sample`s with the same URL is perfectly fine to have on the DB and then we'd possibly associate the imported `LabeledSample` with some unrelated existing `Sample`.

There are two main cases right now where we'd like to import samples with this API:

1. Samples that were exported and then fixed by hand. These use the `LabeledSample` id as their filename, so we have a source of information that tells us exactly which `LabeledSample` the imported one will supersede, and from that we can get the corresponding `Sample`. In order to get this `id` we'll have to send it in the API request as I don't think you can get the filename of the attached file, just its contents.
2. Samples that were labeled in past projects using Fathom and did not use FTA. For those there could be existing `Sample`s with that URL (and associated `LabeledSample`s), but we shouldn't associate them with existing database entries as they don't have any relation.

Does that make sense? Sorry for not catching this earlier. Hopefully this function will end up being simpler.

Reply:
If I'm understanding correctly

1. Using the filename of the imported file(will need to be added) and URL (+ enough info to ensure uniqueness) determine if an existing LabeledSample exists and use its corresponding Sample.
2. If there is no matching LabeledSample found create a new Sample and LabeledSample

#1 would replace the existing check for the URL.

Reply:
The filename should be enough on its own to ensure uniqueness as it's an id. There should be no need to match on the URL. In case we don't have a filename/id we should just skip this process completely and always create a new `Sample` and `LabeledSample`.

Reply:
ok, implemented relying on Ids only